src/main/kotlin/tf/sou/mc/pal/utils/extensions.kt

```diff
@@ -105,7 +105,7 @@ fun Inventory.countAvailableSpace(item: Material): Int {
 /**
  * Attempts to find items in an inventory that differ from the provided [Material].
  */
-fun Inventory.findBadItems(allowed: Material): List<ItemStack> {
+fun Inventory.findBadItems(allowed: Material): List<ItemStack?> {
```

Comment:
The actual problem is the method body here. At some point, PaperMC changed the signature of `Inventory.getContents` to `@Nullable ItemStack @NotNull [] getContents()`, preventing the use of a simple `.filter` (since Kotlin can't infer List<T> from a closure like `.filter { it != null }`) call on the returned contents. You are better off changing the body to this:


```kotlin
fun Inventory.findBadItems(allowed: Material): List<ItemStack> {
    val filteredInventory = mutableListOf<ItemStack>()
    for (item in contents) {
        if (item != null && item.type != allowed) filteredInventory.add(item)
    }
    return filteredInventory
}
```